scipy/optimize/_constraints.py

```diff
@@ -300,12 +333,8 @@ def new_bounds_to_old(lb, ub, n):
     If any of the entries in lb/ub are -np.inf/np.inf they are replaced by
     None.
     """
-    lb = np.asarray(lb)
-    ub = np.asarray(ub)
-    if lb.ndim == 0:
-        lb = np.resize(lb, n)
-    if ub.ndim == 0:
-        ub = np.resize(ub, n)
+    lb = np.broadcast_to(lb, n)
```

Comment:
See the above line:

> If any of the entries in lb/ub are -np.inf/np.inf they are replaced by None.

(indicating that the default thought is towards -np.inf/np.inf

Reply:
I'm not sure I understand. I wrote this originally, and I don't think the `None` here a hint at a default. It is `None` because that (unfortunately) is the standard old-style way of representing either positive or negative infinity.

![image](https://user-images.githubusercontent.com/6570539/147629132-b12f1ffd-4c89-424d-9bcb-3f74f6a9e0f5.png)